src/main/java/com/khutircraftubackend/product/ProductController.java

```diff
@@ -13,69 +15,80 @@
+    @ResponseStatus(HttpStatus.OK)
+    public ProductResponse patchProduct(
+            @Valid
+            @PathVariable Long productId,
+            @ModelAttribute ProductUpdateRequest request,
+            @RequestPart(value = "thumbnailImage", required = false) MultipartFile thumbnailImage,
+            @RequestPart(value = "image", required = false) MultipartFile image) throws IOException {
 
         ProductEntity patchedProduct = productService.patchProduct(productId, request, thumbnailImage, image);
 
-        return ResponseEntity.ok(patchedProduct);
+        return productMapper.toProductResponse(patchedProduct);
     }
 
     @PutMapping("/{productId}")
     @PreAuthorize("hasRole('SELLER') and @productService.canModifyProduct(#productId)")
-    public ResponseEntity<ProductEntity> updateProduct(@PathVariable Long productId,
-                                                       @ModelAttribute ProductUpdateRequest request,
-                                                       @RequestParam("thumbnailImage") MultipartFile thumbnailImage,
-                                                       @RequestParam("images") MultipartFile image) throws IOException {
+    @ResponseStatus(HttpStatus.OK)
+    public ProductResponse updateProduct(
+            @Valid
+            @PathVariable Long productId,
+            @ModelAttribute ProductUpdateRequest request,
+            @RequestPart(value = "thumbnailImage", required = false) MultipartFile thumbnailImage,
+            @RequestPart(value = "image", required = false) MultipartFile image) throws IOException {
+
         ProductEntity updatedProduct = productService.updateProduct(productId, request, thumbnailImage, image);
 
-        return ResponseEntity.ok(updatedProduct);
+        return productMapper.toProductResponse(updatedProduct);
     }
 
     @DeleteMapping("/{productId}")
     @PreAuthorize("hasRole('SELLER') and @productService.canModifyProduct(#productId)")
-    public ResponseEntity<Void> deleteProduct(@PathVariable Long productId) throws IOException {
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteProduct(@PathVariable Long productId) throws IOException {
+
         productService.deleteProduct(productId);
-        return ResponseEntity.noContent().build();
+
     }
 
     @DeleteMapping("/delete-all")
     @PreAuthorize("hasRole('SELLER')")
-    public ResponseEntity<Void> deleteAllProductsForCurrentSeller() throws IOException {
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteAllProductsForCurrentSeller() throws IOException {
+
         productService.deleteAllProductsForCurrentSeller();
-        return ResponseEntity.noContent().build();
+
     }
 
     @GetMapping
     public List<ProductEntity> getAllProducts(
             @RequestParam(defaultValue = "0") int offset,
             @RequestParam(defaultValue = "4") int limit) {
+
         return productService.getProducts(offset, limit);
```

Comment:
do we need to map it to ProductResponse ?

Reply:
I don't understand why.

Reply:
but we return DTO from other methods, like `createProduct()`, `updateProduct()`, `patchProduct()`.
`getAllProducts` is just another method, but only difference is - it returns a Collection of something.
why not Collection of DTO objects? if we don't care about DTO here, why do we use DTOs in all other places like create-, update- patch- Product methods?

let's be consistent. 
If it's ok for us to expose ProductEntity - return it everywhere, otherwise always return DTO as response.

Reply:
Fixed by analogy with the category.